test/net/sourceforge/kolmafia/request/NPCPurchaseRequestTest.java

```diff
@@ -2,25 +2,49 @@
 
 import static internal.helpers.HttpClientWrapper.getRequests;
 import static internal.helpers.Networking.assertPostRequest;
+import static internal.helpers.Networking.html;
 import static internal.helpers.Player.withEquippableItem;
 import static internal.helpers.Player.withEquipped;
+import static internal.helpers.Player.withMeat;
+import static internal.helpers.Player.withNextResponse;
 import static internal.helpers.Player.withPath;
+import static internal.helpers.Player.withQuestProgress;
 import static org.hamcrest.MatcherAssert.assertThat;
 import static org.hamcrest.Matchers.equalTo;
 import static org.hamcrest.Matchers.hasSize;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
 
 import internal.helpers.Cleanups;
 import internal.helpers.HttpClientWrapper;
 import net.sourceforge.kolmafia.AscensionPath;
+import net.sourceforge.kolmafia.KoLCharacter;
+import net.sourceforge.kolmafia.KoLConstants;
+import net.sourceforge.kolmafia.KoLmafia;
+import net.sourceforge.kolmafia.objectpool.ItemPool;
+import net.sourceforge.kolmafia.persistence.QuestDatabase;
+import net.sourceforge.kolmafia.preferences.Preferences;
 import net.sourceforge.kolmafia.session.EquipmentManager;
+import net.sourceforge.kolmafia.session.MallPriceManager;
+import org.junit.jupiter.api.AfterEach;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Nested;
 import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.CsvSource;
 
 class NPCPurchaseRequestTest {
   @BeforeEach
   public void beforeEach() {
     HttpClientWrapper.setupFakeClient();
+
+    KoLCharacter.reset("NPCPurchaseTest");
+    Preferences.reset("NPCPurchaseTest");
+  }
+
+  @AfterEach
+  public void afterEach() {
+    KoLCharacter.reset(true);
```

Comment:
we've been trying to move stuff like this into `Cleanups` to avoid expensive resets between tests.

Can you make new cleanups for:
```
withNPCStoreReset();
withCanInteract(boolean interact);
```

and shift the tests to use them?